flow.cylc

```diff
@@ -0,0 +1,20 @@
+[scheduler]
+    allow implicit tasks = True
+[scheduling]
+    [[graph]]
+        R1 = """
+            setup_environment => preprocess
+        """
+[runtime]
+    [[setup_environment]]
+        script = """
+            python3.11 -m venv "$CYLC_WORKFLOW_SHARE_DIR/venv"
+            source "$CYLC_WORKFLOW_SHARE_DIR/venv/bin/activate"
+            pip install "$CYLC_WORKFLOW_RUN_DIR"
```

Comment:
Why create an environment for this use case within a container? Wouldn't things work in the base Python installation? Are there any risks?

Reply:
You could run it in the base python environment, if you were only running it once and using the container as the environment. In this case, I want to create a separate python environment for each run, so that it's easier to be sure that we're only using the dependencies listed in the `pyproject.toml` file. In the devcontainer I've got a few different environments – one for each run, plus one where I test the code without cylc.

For the version I used to run the processing for the lab last week, I dropped explicitly creating the venv and used pipx to set up fsdproc.